In [1]:
import pandas as pd
df = pd.read_csv("allbacks.csv")

df

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb
5,6,641,367,600,hb
6,7,1228,396,1075,hb
7,8,412,0,250,pb
8,9,953,0,700,pb
9,10,929,0,650,pb


Sort the values of "cover" column and call it a "sorted_series"

In [2]:
sorted_series = df["cover"].sort_values()

In [3]:
print(sorted_series)

0     hb
1     hb
2     hb
3     hb
4     hb
5     hb
6     hb
7     pb
8     pb
9     pb
10    pb
11    pb
12    pb
13    pb
14    pb
Name: cover, dtype: str


Find the middle index by performing floor division and use that index to find the median and store it in "median_cover"

In [4]:
median_cover = sorted_series.iloc[len(sorted_series) // 2]

In [5]:
print(median_cover)

pb


The median cover is pb (paperback), which happens to be the middle value in the sorted cover list. 

Find the mean weight of all books

In [6]:
print(df["weight"].mean())

706.6666666666666


The mean weight of all books is about 706.67 grams.

Find the mean weight within each cover type (hardcover vs. paperback)

In [7]:
df.loc[df["cover"] == "hb", "weight"].mean()

np.float64(796.4285714285714)

In [8]:
df.loc[df["cover"] == "pb", "weight"].mean()

np.float64(628.125)

In [9]:
df.groupby(by="cover", dropna=False).mean()

,rownames,volume,area,weight
cover,,,,
hb,4.0,833.571429,391.714286,796.428571
pb,11.5,855.500000,0.000000,628.125000


The mean weight for hardbacks is 796.429 grams. The mean weight for paperbacks is 628.125 grams. I can clearly see that the paperbacks weigh significantly less than the hardbacks, with paperbacks' mean weight weighing less than the overall mean weight for all books.

Getting the mean of all hardbacks by hand:

(800 + 950 + 1050 + 350 + 750 + 600 + 1075) / 7 = 796.42857143

It verifies with python's calculation!

Repo address: https://github.com/yeonghwang-hub/cs82a-portfolio